sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1227,16 +1229,27 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.24
 
-    monotonic_cst : array-like of int of shape (n_features), default=None
-        Indicates the monotonic constraint to enforce on each feature.
-          - 1: monotonic increase
-          - 0: no constraint
-          - -1: monotonic decrease
+    monotonic_cst : array-like of int of shape (n_features) or dict, default=None
+        Monotonic constraint to enforce on each feature are specified using the
+        following integer values:
 
+        - 1: monotonic increase
+        - 0: no constraint
+        - -1: monotonic decrease
+
+        If a dict with str keys, map feature to monotonic constraints by name.
+        If an array, the feature are mapped to constraints by position. See
+        :ref:`monotonic_cst_features_names` for a usage example.
```

Comment:
The reference works as expected:

https://output.circle-artifacts.com/output/job/bc0fc35c-0030-48fe-a5b5-5c6f3fe752fd/artifacts/0/doc/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html